src/evict/evict_page.c

```diff
@@ -142,6 +142,29 @@ done:	if (((inmem_split && ret == 0) || (forced_eviction && ret == EBUSY)) &&
 
 	return (ret);
 }
+/*
+ * __evict_ref_delete --
+ *	Mark a reference deleted.
+ */
+static int
+__evict_ref_delete(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+	WT_PAGE *parent;
+	WT_PAGE_INDEX *pindex;
+	uint32_t deleted_entries;
+
+	WT_PUBLISH(ref->state, WT_REF_DELETED);
+	if (__wt_ref_is_root(ref))
+		return (0);
+
+	parent = ref->home;
+	WT_INTL_INDEX_GET(session, parent, pindex);
+	deleted_entries = __wt_atomic_addv32(&pindex->deleted_entries, 1);
+	if (deleted_entries > pindex->entries / 10)
+		WT_RET(__wt_split_reverse(session, parent->pg_intl_parent_ref));
+
+	return (0);
+}
```

Comment:
@michaelcahill, I'd leave the set of `WT_REF_DELETED` in the callers and move the `WT_WITH_PAGE_INDEX` work into `__evict_ref_delete()`. That keeps the work to set the fields of the `WT_REF` in one place, rather than splitting it into two parts, and puts the page-index work in a single place as well.